enterprise/com/src/main/java/org/neo4j/com/storecopy/StoreCopyClient.java

```diff
@@ -281,6 +283,20 @@ public boolean visit( CommittedTransactionRepresentation transaction ) throws IO
             // header of the log that we just wrote.
             File currentLogFile = logFiles.getLogFileForVersion( currentLogVersion );
             writeLogHeader( fs, currentLogFile, currentLogVersion, max( BASE_TX_ID, endTxId - 1 ) );
+
+            if ( !forensics )
+            {
+                // since we just create new log and put checkpoint into it with offset equals to
+                // LOG_HEADER_SIZE we need to update last transaction offset to be equal to this newly defined max
+                // offset otherwise next checkpoint that use last transaction offset will be created for non
+                // existing offset that is in most of the cases bigger than new log size.
+                // Recovery will treat that as last checkpoint and will not try to recover store till new
+                // last closed transaction offset will not overcome old one. Till that happens it will be
+                // impossible for recovery process to restore the store
+                File neoStore = new File( tempStoreDir, NeoStore.DEFAULT_NAME );
+                NeoStore.setRecord( fs, neoStore, NeoStore.Position.LAST_CLOSED_TRANSACTION_LOG_BYTE_OFFSET,
```

Comment:
* if `forensics` option is true - we copy logs together with storage so value of `NeoStore.Position.LAST_CLOSED_TRANSACTION_LOG_VERSION` should be still valid
* if `forensics` is false we do not copy logs and creating new blanc log file instead with only one checkpoint in it with offset = LOG_HEADER_SIZE, updating `NeoStore.Position.LAST_CLOSED_TRANSACTION_LOG_VERSION` with LOG_HEADER_SIZE should only cause creation of new checkpoint with same offset only in case it will be created. This new offset will not over exceed new log file size and will not cause out of sequence checkpoint records in a file. (Ex. record 1: Checkpoint with offset LOG_HEADER_SIZE,  record 2: checkpoint with offset 0 )